src/test/java/lrbg/codriver/parser/ParserTest.java

```diff
@@ -0,0 +1,158 @@
+package lrbg.codriver.parser;
+
+import lrbg.codriver.command.ByeCommand;
+import lrbg.codriver.command.DeadlineCommand;
+import lrbg.codriver.command.DeleteCommand;
+import lrbg.codriver.command.EventCommand;
+import lrbg.codriver.command.ListCommand;
+import lrbg.codriver.command.MarkCommand;
+import lrbg.codriver.command.TodoCommand;
+import lrbg.codriver.command.UnknownCommand;
+import lrbg.codriver.command.UnmarkCommand;
+import lrbg.codriver.command.FindCommand;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * Test Class for Parser.
+ */
+public class ParserTest {
+    @Test
+    public void parseTest_correctArgument_success() {
+        try {
+            assertTrue((new TodoCommand("1234"))
+                    .testEquals(Parser.parse("todo 1234")));
+            assertTrue((new DeadlineCommand("1234", Parser.parseDate("2021-08-25")))
+                    .testEquals(Parser.parse("deadline 1234 /by 2021-08-25")));
+            assertTrue((new EventCommand("1234",
+                    Parser.parseDate("2021-08-25"), Parser.parseDate("2021-08-26")))
+                    .testEquals(Parser.parse("event 1234 /from 2021-08-25 /to 2021-08-26")));
+            assertTrue((new ByeCommand()).testEquals(Parser.parse("bye")));
+            assertTrue((new UnknownCommand("blah")).testEquals(Parser.parse("blah blah blah")));
+            assertTrue((new ListCommand()).testEquals(Parser.parse("list")));
+            assertTrue((new DeleteCommand(1)).testEquals(Parser.parse("delete 1")));
+            assertTrue((new MarkCommand(1)).testEquals(Parser.parse("mark 1")));
+            assertTrue((new UnmarkCommand(1)).testEquals(Parser.parse("unmark 1")));
+            assertTrue((new FindCommand("1234")).testEquals(Parser.parse("find 1234")));
+        } catch (Exception e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void parseTest_deadlineIncorrectArgument_exceptionThrown() {
```

Comment:
Might be better if our deadlineIncorrectArgument has a more specific case. There are many ways in which the argument is incorrect. when i run the test in my terminal, i should be able to know what kind of incorrect argument throws an error!